Issue/woomob 884 shipping labels add units for dimension and weight fields

JorgeMucientes opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->

Closes: WOOMOB-884

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
The new shipping labels UI was displaying numbers such as weight, height, or price without the corresponding units. This PR resolves that.

### Testing information
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->
1. Open an order eligible for shipping labels
2. Tap on select a package and check that the input text to add a new package contain the correct units (likely "cm")
3. Navigate back. 
4. Open customs form
5. Check that product details in collapsed mode show price per unit and weight with the correct unit. 
6. Expand the product and check that the input text fields now have a currency symbol and unit for price and weight

| Before  | After |
| ------------- | ------------- |
| <img width="300"  src="https://github.com/user-attachments/assets/6b6f4ef3-8871-43b7-89f7-52f00e29abc2">   | <img width="300"  src="https://github.com/user-attachments/assets/ab77cbf7-26b3-4bb6-b7ed-1e2576a13b75">  |
| <img width="300"  src="https://github.com/user-attachments/assets/556fdc95-124b-4cfc-99fa-5b67108b7dfe">  | <img width="300"  src="https://github.com/user-attachments/assets/decaa71d-b4e3-46ea-9dd8-823cc5e7c67e">  |

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

JorgeMucientes committed:
Update git ignore to include .junie folder

JorgeMucientes committed:
Display dimension unit in package creation screen

This commit introduces the display of dimension units (e.g., "cm") next to the length, width, and height input fields in the custom package creation screen.

JorgeMucientes committed:
Manually add length, width, height labels

JorgeMucientes committed:
Use dimension unit from fetched store options

JorgeMucientes committed:
Passes store options to customs form to display missing units

JorgeMucientes committed:
Add missing weight unit and price symbol to collapsed item details

Customs form was missing the currency symbol as well as weight unit when displaying product item details in collapsed mode

JorgeMucientes commented on .gitignore:
```diff
@@ -116,4 +116,7 @@ google-upload-credentials.json
 
 # Cursor AI Editor
 /.cursor/
+
+# Junie AI Assistant
+.junie/
```
This is unrelated to the PR, but I started using Jetbrains Junie and had to add this in order to avoid tracking accidentally my *.md config files inside the .junie/ folder.

dangermattic commented:
<!--
  0 Errors
  2 Warnings: This PR is larger than 300 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/263">22.9</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

JorgeMucientes committed:
Update formattedPriceAndWeight value when input changes

JorgeMucientes committed:
Remove unnecessary mocks

JorgeMucientes committed:
Fix unit tests

JorgeMucientes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/packages/ui/WooShippingCustomPackageScreen.kt:
```diff
@@ -99,41 +101,83 @@ fun WooShippingCustomPackageCreationScreen(
-                    modifier = modifier.weight(1f)
-                )
+                Column(modifier = modifier.weight(1f)) {
+                    Text(
+                        text = stringResource(id = R.string.woo_shipping_labels_package_creation_length),
+                        style = MaterialTheme.typography.body2,
+                        color = MaterialTheme.colors.onSurface.copy(alpha = 0.6f),
+                    )
+                    WCOutlinedTextField(
+                        value = packageLength,
+                        onValueChange = onLengthChange,
+                        label = "",
+                        trailingIcon = {
+                            Text(
+                                text = dimensionUnit,
+                                style = MaterialTheme.typography.body2,
+                                color = MaterialTheme.colors.onSurface.copy(alpha = 0.6f)
+                            )
+                        },
+                        singleLine = true,
+                        keyboardOptions = KeyboardOptions(
+                            keyboardType = KeyboardType.Number,
+                            imeAction = ImeAction.Next
+                        ),
+                        modifier = Modifier.fillMaxWidth()
+                    )
```
I had to slightly change the UI here and extract the "label" value from `WCOutlinedTextField` to an outside `Text`. The reason is that the `trailingIcon` with the unit and the `label` would overlap each other leading to a weird new line effect on the input text. Example:



https://github.com/user-attachments/assets/924b9998-ebfb-4c7c-89d4-04c684895a5b




This also happens in the customs form for value and weight per unit. However, the issue is harder to reproduce as the input text is always pre-filled with the product data.

JorgeMucientes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/packages/ui/WooShippingCustomPackageScreen.kt:
```diff
@@ -43,6 +43,7 @@ fun WooShippingCustomPackageCreationScreen(viewModel: WooShippingLabelPackageCre
         packageLength = viewState?.customPackageCreationData?.length.orEmpty(),
         packageWidth = viewState?.customPackageCreationData?.width.orEmpty(),
         packageWeight = viewState?.customPackageCreationData?.weight.orEmpty(),
+        dimensionUnit = viewState?.packagesData?.storeOptions?.dimensionUnit.orEmpty(),
```
Because store options are fetched from the API everytime, there's some delay for the unit to show. It's not a big deal so I kept it that way. Alternatively we can pass store options as fragment args to the screen to avoid that effect. Let me know what you think:

https://github.com/user-attachments/assets/52616fdb-e66d-4cca-84a3-0bbc1b5a5c6e

irfano replied:
> Alternatively we can pass store options as fragment args to the screen to avoid that effect. Let me know what you think:

I think this would be better. You’ve already done that for `WooShippingLabelCustomsFormFragment`. I agree it’s not a big deal, but the solution is also easy, unless I’m missing something. Just sharing my opinion; I don’t see this as a blocker.

JorgeMucientes replied:
Done!

irfano commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/customs/WooShippingCustomsFormViewModel.kt:
```diff
@@ -30,16 +31,21 @@ import javax.inject.Inject
+        initialValue = ViewState(
+            currencySymbol = storeOptions.currencySymbol,
+            weightUnit = storeOptions.weightUnit
+        )
```
This part was changed in the `release/22.8` branch. It’s better to wait for the backmerge PR before merging this into trunk.

irfano reviewed (changes requested):
The new UI changes look great. I couldn’t find any issues, even after increasing the font size.

I added some comments, but there’s only a single minor blocker related to the theme color:

The units are not readable in dark mode on the customs screen.

<img width="400" src="https://github.com/user-attachments/assets/c0c895d3-6d0c-4af2-b756-05baf2f797db" />

JorgeMucientes committed:
Use Material theme instead of Material 3 for customs form

JorgeMucientes committed:
Pass store options to package screen

When entering package creation screen, displayed units like "cm" take a few seconds to be displayed because store options are fetched from the API. Passing the storeOptions from shipping label creation screen provides a better experience.

JorgeMucientes commented:
Thanks for the review @irfano 

> The units are not readable in dark mode on the customs screen.

Good catch, I think the main issue here was that we were using the Material3 theme and Texstyles instead of using the ones we have defined for the whole app. I replaced the text style usages of material3 and set the correct ones. That fixes the dark theme color issues. 

Ready for another round!

irfano reviewed (approved):
Thanks for addressing my feedback. It works great. LGTM! 👍🏻

irfano committed:
Fix WooShippingLabelPackageCreationViewModelTest.kt

irfano commented:
I added a fix(https://github.com/woocommerce/woocommerce-android/pull/14371/commits/3162a3feaf7818a3df97c93e695f07fe29ded020) for the failing unit test.

JorgeMucientes committed:
Merge branch 'trunk' into issue/woomob-884-shipping-labels-add-units-for-dimension-and-weight-fields

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/WooShippingLabelCreationViewModel.kt
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/customs/WooShippingCustomsFormViewModel.kt
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/customs/products/WooShippingCustomsProductListItem.kt
#	WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/customs/WooShippingCustomsFormViewModelTest.kt

JorgeMucientes committed:
Fix some issues after merging

JorgeMucientes committed:
Fix failing unit test resulting from bad merge
